ding/world_model/base_world_model.py

```diff
@@ -334,9 +331,9 @@ def rollout(self, obs: Tensor, actor_fn: Callable[[Tensor], Tuple[Tensor, Tensor
             torch.stack(obss),
             torch.stack(actions),
             # rewards is an empty list when horizon=0
-            torch.stack(rewards) if rewards else torch.tensor(rewards, device=device),
+            torch.stack(rewards) if rewards else torch.tensor(rewards, device=obs.device),
```

Comment:
modify it to a more clear condition, `if xxx` can be a little confuse here

Reply:
I can change to ` torch.stack(rewards) if rewards != [] else torch.tensor(rewards, device=obs.device)`, but I do not think it is necessary.

Reply:
ok, just add more comments about this

Reply:
sure